src/entities/selectionBox.ts

```diff
@@ -0,0 +1,125 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Entity {
+  export class SelectionBox {
+    private _parentNode: Element;
+    private _box: D3.Selection;
+    private _core: D3.Selection;
+    private _edgeT: D3.Selection;
+    private _edgeB: D3.Selection;
+    private _edgeL: D3.Selection;
+    private _edgeR: D3.Selection;
+    private _cornerTL: D3.Selection;
+    private _cornerTR: D3.Selection;
+    private _cornerBL: D3.Selection;
+    private _cornerBR: D3.Selection;
+
+    private _topLeft: Point = { x: 0, y: 0 };
+    private _bottomRight: Point = { x: 0, y: 0 };
+
+    private _detectionRadius = 2;
+
+    constructor(parent: D3.Selection) {
+      this._parentNode = parent.node();
+      this._box = parent.append("g").classed("selection-box", true).remove();
+      this._core = this._box.append("rect").classed("selection-area", true);
+
+      var createLine = () => this._box.append("line").style({
+                               "opacity": 0,
+                               "stroke": "pink",
+                               "stroke-width": 2 * this._detectionRadius
+                             });
+      this._edgeT = createLine().classed("selection-edge-tb", true);
+      this._edgeB = createLine().classed("selection-edge-tb", true);
+      this._edgeL = createLine().classed("selection-edge-lr", true);
+      this._edgeR = createLine().classed("selection-edge-lr", true);
+
+      var createCorner = () => this._box.append("circle")
+                                   .attr("r", this._detectionRadius)
+                                   .style({
+                                     "opacity": 0,
+                                     "fill": "pink"
+                                   });
+      this._cornerTL = createCorner().classed("selection-corner-tl", true);
+      this._cornerTR = createCorner().classed("selection-corner-tr", true);
+      this._cornerBL = createCorner().classed("selection-corner-bl", true);
+      this._cornerBR = createCorner().classed("selection-corner-br", true);
```

Comment:
Do we want these classes to be on by default?  I'm thinking that it is switched off by default and DragInteraction for example would turn these on